src/components/ColorInput.jsx

```diff
@@ -1,9 +1,11 @@
 import React from "react"
-
-const ColorInput = ({ inputId, bgColor, onChange }) => {
+import {AiOutlineCloseCircle} from 'react-icons/ai'
+const ColorInput = ({ inputId,colorsLength, bgColor, onChange,handleDelete }) => {
+  console.log(typeof inputId)
   return (
     <div className="colorInputContainer">
       <div className="colorSwatch" style={{ backgroundColor: bgColor }}></div>
+      {(colorsLength > 2)?<button className="deleteButton"><AiOutlineCloseCircle color="grey"  onClick={()=>handleDelete(inputId-1)}/></button>:null}
```

Comment:
Please move this line of code to line 19 after the input element so it is inside the input container.

Also, the onClick handler needs moving from the icon to the button. 
```suggestion
      {(colorsLength > 2)?<button className="deleteButton" onClick={()=>handleDelete(inputId-1)}><AiOutlineCloseCircle color="grey"  /></button>:null}
```

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

Reply:
Oh I nearly forgot - this button also needs an aria-label="delete input" as it doesn't have any discernable text that could be read aloud by a screen reader.

Reply:
okay I will start working on the changes.

Reply:
**Can you answer these question. It will make a lot of things clear to me**
- What is the best way to keep my fork updated with the main repo?
- Should I always make the new branch from the main branch (According to my knowledge this is the only way to all the branch separated)?
- Why do we need aria-label(what is its functionality)?

Reply:
At the top of your forked repo you should see a bar that looks like this:
<img width="674" alt="image" src="https://user-images.githubusercontent.com/57045550/177013167-79279d7c-7b57-488c-b3af-514001f20787.png">
If you click on fetch upstream it'll sync the two repositories. Then in your local clone you do git pull in the main branch. Then you change into your working branch and do git merge main and finally git push to sync your branch with your fork.

Yes, always make a new branch from main branch. You won't see some of the changes you made on another branch until the PR has been merged but it's the best way to avoid conflicts.

Because the button doesn't contain any text when a screen reader uses the site all they will hear is 'button' which doesn't give them any context about what the button does. If we add an aria label that will also be read out so they will instead hear 'button, delete input' and that will give them context about what happens when they click on the button.

Hope that helps!

Reply:
Yeah it made alot of sense. Thanks for the explanation 👍🙂